components/authz-service/server/v2/project.go

```diff
@@ -158,9 +163,22 @@ func (s *state) UpdateProject(ctx context.Context,
 }
 
 func (s *state) ApplyRulesStart(
-	context.Context, *api.ApplyRulesStartReq) (*api.ApplyRulesStartResp, error) {
+	ctx context.Context, _ *api.ApplyRulesStartReq) (*api.ApplyRulesStartResp, error) {
 	s.log.Info("apply project rules: START")
-	err := s.projectUpdateManager.Start()
+
+	err := s.store.ApplyStagedRules(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal,
+			"error applying staged projects: %s", err.Error())
+	}
+
+	err = s.policyRefresher.Refresh(ctx)
```

Comment:
Uhm I don't get it. What hinders us from refreshing in the store's `ApplyStagedRules` method?

[Looks like we do that](https://github.com/chef/automate/pull/655/commits/b45ee71418d0c8325fc65162db54f088949f3889#diff-d28b0cc9c342bf1da873bddba25960f6R1347). I don't think I understand why this is necessary then.

Reply:
Because that method is async whereas this is sync. You run into race conditions if you don't use this method.

Reply:
`notifyPolicyChange` is async. It doesn't wait until the OPA cache is updated. Thus, we need the `policyRefresher.Refresh` call, because we need the OPA cache to be updated before `projectUpdateManager.Start`.

Reply:
@tylercloke you mean the gocache . . .right? I still don't think we need to bump the OPA cache unless a project is deleted-- and we _should_ probably worry about that in addition to rules changing, but we don't currently. . . 😳

that said, I know we've woven the OPA cache bumping with the gocache so it's kinda moot.

Reply:
assuming i'm remembering things correctly from our conversation:
there are 2 parts to this dance. 1 is updating the database, the other is synchronizing the OPA store with the database. Updating the database will at some point in the future trigger replication. This is asynchronous. If you want to change the database and update the OPA store, you need to call refresh on the policy refresher thing because the database layer doesn't know about the OPA layer.